dbt/dbt-run.py

```diff
@@ -133,10 +146,90 @@ def run_incremental_models():
   # update incremental models (and tables if there are any)
   subprocess.run(["dbt", "run",  "--profiles-dir", ".dbt", "--exclude", "config.materialized:view"])
 
+def get_pending_doc_count():
+  with connection() as conn:
+      with conn.cursor() as cur:
+          cur.execute(f"""
+              SELECT SUM(pending)
+              FROM {os.getenv('POSTGRES_SCHEMA')}.couchdb_progress
+          """)
+          return cur.fetchone()[0]
+
+def update_batch_status(timestamp, status):
+  with connection() as conn:
+    with conn.cursor() as cur:
+      # insert new entry
+      cur.execute(
+        f"INSERT INTO {os.getenv('POSTGRES_SCHEMA')}.dbt_batch_status (timestamp, status) VALUES (%s, %s);", [timestamp, status]
+      )
+      conn.commit()
+
+def get_last_processed_timestamp():
+  with connection() as conn:
+    with conn.cursor() as cur:
+      cur.execute(f"""
+          SELECT MAX(timestamp)
+          FROM {os.getenv('POSTGRES_SCHEMA')}.dbt_batch_status
+          WHERE status = 'success'
+      """)
+      result = cur.fetchone()
+      if result and result[0]:
+        return result[0]
+      return '1970-01-01 00:00:00'
+    
+def get_max_timestamp():
+  with connection() as conn:
+    with conn.cursor() as cur:
+      cur.execute(f"""
+          SELECT MAX(saved_timestamp)
+          FROM {os.getenv('POSTGRES_SCHEMA')}.document_metadata
+      """)
+      return cur.fetchone()[0]
+    
+def handle_batch_update_error(last_processed_timestamp, dataemon_interval):
+   print(f"Error running dbt for batch {last_processed_timestamp}")
+   update_batch_status(last_processed_timestamp, "error")
+   time.sleep(dataemon_interval)
+
+def handle_batch_update_success(last_processed_timestamp, dataemon_interval, max_timestamp):
+  update_batch_status(last_processed_timestamp, "success")
+
+  if max_timestamp == last_processed_timestamp:
+     time.sleep(dataemon_interval)
+   
+def run_dbt_in_batches():
+  last_processed_timestamp = get_last_processed_timestamp()
+  batch_size = int(os.getenv("DBT_BATCH_SIZE") or 10000)
+  dataemon_interval = int(os.getenv("DATAEMON_INTERVAL") or 5)
+
+  while True:
+     update_dbt_deps()
+     result = subprocess.run([
+        "dbt", "run",
+        "--profiles-dir", ".dbt",
+        "--vars", f'{{start_timestamp: "{last_processed_timestamp}", batch_size: {batch_size}}}',
+        "--exclude", "config.materialized:view"
```

Comment:
What does this exclusion mean?

Reply:
Ok, so we exclude incremental models when we do the batch run, but do we trigger their refresh when our batched runs are done?

Reply:
Ah, I see, so we always pass this parameter. But this doesn't end up excluding materialized views?

Reply:
No. The dbt run command is not required to refresh the data, see docs [here](https://docs.getdbt.com/docs/build/materializations#scheduled-refreshes).